lib/ansible/modules/cloud/amazon/sqs_queue.py

```diff
@@ -139,178 +217,184 @@
 '''
 
 import json
-import traceback
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict, compare_aws_tags, snake_dict_to_camel_dict
 
 try:
-    import boto.sqs
-    from boto.exception import BotoServerError, NoAuthHandlerFound
-    HAS_BOTO = True
-
+    from botocore.exceptions import BotoCoreError, ClientError, ParamValidationError
 except ImportError:
-    HAS_BOTO = False
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
-
-
-def create_or_update_sqs_queue(connection, module):
-    queue_name = module.params.get('name')
+    pass  # handled by AnsibleAWSModule
+
+ARGUMENT_SPEC = dict(
+    state=dict(type='str', default='present', choices=['present', 'absent']),
+    name=dict(required=True, type='str'),
+    queue_type=dict(type='str', default='standard', choices=['standard', 'fifo']),
+    delay_seconds=dict(type='int', aliases=['delivery_delay']),
+    maximum_message_size=dict(type='int'),
+    message_retention_period=dict(type='int'),
+    policy=dict(type='dict'),
+    receive_message_wait_time_seconds=dict(type='int', aliases=['receive_message_wait_time']),
+    redrive_policy=dict(type='dict'),
+    visibility_timeout=dict(type='int', aliases=['default_visibility_timeout']),
+    kms_master_key_id=dict(type='str'),
+    kms_data_key_reuse_period_seconds=dict(type='int', aliases=['kms_data_key_reuse_period']),
+    content_based_deduplication=dict(type='bool'),
+    tags=dict(type='dict'),
+    purge_tags=dict(type='bool', default=False),
+)
+NON_ATTRIBUTES = ('name', 'state', 'queue_type', 'tags', 'purge_tags')
+COMPATABILITY_KEYS = dict(
```

Comment:
These should be handled in the arg_spec using `aliases`.

Reply:
They're used more to ensure the return keys/values are what we expect.

I've updated it so that we *add* the old return values but document the return values that match new keys.